inlang/source-code/sdk2/src/project/loadProjectFromDirectory.ts

```diff
@@ -28,27 +28,73 @@ export async function loadProjectFromDirectoryInMemory(
 	});
 	await copyFiles({ fs: args.fs, path: args.path, lix: tempProject.lix });
 
-	// TODO loadMessages - call tempProject.lix.settled() to wait for the new settings file
-
-	// TODO loadMessages - since settings are not reactiv yet - just reload the project
+	// TODO call tempProject.lix.settled() to wait for the new settings file, and remove reload of the proejct as soon as reactive settings has landed
+	// NOTE: we need to ensure two things:
+	// 1. settled needs to include the changes from the copyFiles call
+	// 2. the changes created from the copyFiles call need to be realized and lead to a signal on the settings
 	const project = await loadProjectInMemory({
 		// pass common arguments to loadProjectInMemory
 		...args,
 		blob: await tempProject.toBlob(),
 	});
 
-	for (const plugin of project.plugins.get()) {
-		// TODO loadMessages - make sure that we have configured either loadMessages and saveMessages xor import and export (throw in other cases)
-		if (plugin.loadMessages !== undefined) {
-			await loadLegacyMessages({
-				project,
-				fs: args.fs,
-				loadMessagesFn: plugin.loadMessages,
-			});
+	// TODO i guess we should move this validation logic into sdk2/src/project/loadProject.ts
+	// Two scenarios could arise:
+	// 1. set settings is called from an app - it should detect and reject the setting of settings -> app need to be able to validate before calling set
+	// 2. the settings file loaded from disc here is corrupted -> user has to fix the file on disc
```

Comment:
@samuelstroschein I placed the validation - including corresponding tests - in loadProjectFromDirectory for now. I guess we will move it to the setters of settings together with the changes regarding reactivity. So I would keep that out of scope for [this ticket](https://linear.app/opral/issue/MESDK-163/implement-loadprojectfromdirectory-pluginloadmessages-logic) and think we can merge and close MESDK-163

Reply:
Given that the "one only load and save messages" can be defined is legacay stuff, only used in `loadProjectFromDirectory`, I would keep it in here. 

Other things in a plugin do not need to be validated since every plugin can define any API they need to (no mergnig happens) and we have TypeScript. Validating a few static props of a plugin can be done with TypeScript. Dynamic function can't be validated anyways.